### PR TITLE
Feature/issue 100 Add option to 'compact' GeoJSON result into single feature

### PR DESCRIPTION
**Pull request by nikki-t**
Github Issue: #100

### Description

Add option to 'compact' GeoJSON result into single feature.

### Overview of work done

- Set OpenAPI query parameters required to match API functionality.
- Defined a new boolean query parameter called 'compact' in the OpenAPI schema.
- Plumbed the 'compact' parameter into the API Lambda function.

  - Returns a 'compact' GeoJSON response for cases where the `Accept=application/json`, `output=geojson`, and `compact=true`.
  - Returns a 'compact' GeoJSON response for cases where the `Accept=application/geo+json`. Default behavior is to return compact response unless `compact` is explicitly set to `false`.

- Updated documentation for the compact GeoJSON response.

### Overview of verification done

Created two new unit tests to test compact response for `application/json` and `application/geo+json` response. All units test pass.

### Overview of integration done

Deployed modifications to SIT environment and ran the following queries.

*Compact application/json*

```bash
curl -v --location 'https://soto.podaac.sit.earthdatacloud.nasa.gov/hydrocron/v1/timeseries?feature=Reach&feature_id=63470800171&start_time=2024-02-01T00:00:00%2b00:00&end_time=2024-10-30T00:00:00%2b00:00&fields=reach_id,time_str,wse&compact=true'
```

```json
{
    "status": "200 OK",
    "time": 2175.824,
    "hits": 2,
    "results": {
        "csv": "",
        "geojson": {
            "type": "FeatureCollection",
            "features": [
                {
                    "id": "0",
                    "type": "Feature",
                    "properties": {
                        "reach_id": [
                            "63470800171",
                            "63470800171"
                        ],
                        "time_str": [
                            "2024-02-01T02:26:50Z",
                            "2024-02-08T13:48:41Z"
                        ],
                        "wse": [
                            "3386.9332",
                            "1453.4136"
                        ],
                        "wse_units": [
                            "m",
                            "m"
                        ]
                    },
                    "geometry": {
                        "type": "LineString",
                        "coordinates": [
                            -45.845445,
                            -16.166559
                        ]
                    }
                }
            ]
        }
    }
}
```

*Coordinates removed

*Compact application/geo+json*

```bash
curl -v --header "Accept: application/geo+json" --location 'https://soto.podaac.sit.earthdatacloud.nasa.gov/hydrocron/v1/timeseries?feature=Reach&feature_id=63470800171&start_time=2024-02-01T00:00:00%2b00:00&end_time=2024-10-30T00:00:00%2b00:00&fields=reach_id,time_str,wse'
```

```json
{
    "type": "FeatureCollection",
    "features": [
        {
            "id": "0",
            "type": "Feature",
            "properties": {
                "reach_id": [
                    "63470800171",
                    "63470800171"
                ],
                "time_str": [
                    "2024-02-01T02:26:50Z",
                    "2024-02-08T13:48:41Z"
                ],
                "wse": [
                    "3386.9332",
                    "1453.4136"
                ],
                "wse_units": [
                    "m",
                    "m"
                ]
            },
            "geometry": {
                "type": "LineString",
                "coordinates": [
                    [
                        -45.845445,
                        -16.166559
                    ]
                ]
            }
        }
    ]
}
```

*Coordinates removed

*Not compact application/geo+json*

```bash
curl -v --header "Accept: application/geo+json" --location 'https://soto.podaac.sit.earthdatacloud.nasa.gov/hydrocron/v1/timeseries?feature=Reach&feature_id=63470800171&start_time=2024-02-01T00:00:00%2b00:00&end_time=2024-10-30T00:00:00%2b00:00&fields=reach_id,time_str,wse&compact=false'
```

```json
{
    "type": "FeatureCollection",
    "features": [
        {
            "id": "0",
            "type": "Feature",
            "properties": {
                "reach_id": "63470800171",
                "time_str": "2024-02-01T02:26:50Z",
                "wse": "3386.9332",
                "wse_units": "m"
            },
            "geometry": {
                "type": "LineString",
                "coordinates": [
                    [
                        -45.845445,
                        -16.166559
                    ]
                ]
            }
        },
        {
            "id": "1",
            "type": "Feature",
            "properties": {
                "reach_id": "63470800171",
                "time_str": "2024-02-08T13:48:41Z",
                "wse": "1453.4136",
                "wse_units": "m"
            },
            "geometry": {
                "type": "LineString",
                "coordinates": [
                    [
                        -45.845445,
                        -16.166559
                    ]
                ]
            }
        }
    ]
}
```

*Coordinates removed.

*Compact GeoJSON - Nodes*

```bash
curl --header "Accept: application/geo+json" --location 'https://soto.podaac.sit.earthdatacloud.nasa.gov/hydrocron/v1/timeseries?feature=Node&feature_id=12228200110861&start_time=2023-11-29T00:00:00Z&end_time=2024-10-30T00:00:00Z&fields=reach_id,node_id,time_str,wse'
```

```json
{
    "type": "FeatureCollection",
    "features": [
        {
            "id": "0",
            "type": "Feature",
            "properties": {
                "reach_id": [
                    "12228200111",
                    "12228200111"
                ],
                "node_id": [
                    "12228200110861",
                    "12228200110861"
                ],
                "time_str": [
                    "2024-01-30T21:19:19Z",
                    "2024-02-06T08:37:09Z"
                ],
                "wse": [
                    "3389.616",
                    "1346.93836"
                ],
                "wse_units": [
                    "m",
                    "m"
                ]
            },
            "geometry": {
                "type": "Point",
                "coordinates": [
                    35.149314,
                    -10.256285
                ]
            }
        }
    ]
}
```

## PR checklist:

* [X] Linted
* [X] Updated unit tests
* [X] Updated changelog
* [X] Integration testing

_See [Pull Request Review Checklist](../CONTRIBUTING.md#reviewing) for pointers on reviewing this pull request_

### CODE REVIEW

**Comment by nikki-t** [comment]
Just an FYI: I created the branch for this PR from the branch for Issue #101 which has not been merged into develop yet so it may cause issues when we merge all open PRs.

**Comment by frankinspace** [comment]
> Just an FYI: I created the branch for this PR from the branch for Issue #101 which has not been merged into develop yet so it may cause issues when we merge all open PRs.

I updated the base for this PR to be feature/issue-101 to reflect the dependency. Once 101 gets merged this PR will automatically get updated to point to develop

File: docs/timeseries.md [hunk 1]
```diff
@@ -2,6 +2,78 @@
...
+
+Possible header and request parameter combinations:
+
+- If the Accept header is `text/csv` or `applicatin/geo+json`, the raw CSV or GeoJSON response is returned.
+- If the Accept header is `application/json` with an output field of `geojson`, the entire JSON object with metadata including GeoJSON response is returned.
+- If the Accept header is `application/json` with an output field of `csv`, the entire JSON object with metadata including CSV response is returned.
+- If the Accept header is `application/json` without an output field, the entire JSON object with metadata including GeoJSON response is returned.
+- If the Accept header is none of the accepted types then a 415 Unsupported is returned.
+
+Example GeoJSON request and response:
```
Comment (frankinspace) [inline docs/timeseries.md:19]: Your descriptions of the code blocks here is reversed. This is the CSV example

**Comment by frankinspace** [comment]
@nikki-t can you work on resolving the merge conflicts?

**Comment by nikki-t** [comment]
@frankinspace - Conflicts resolved and merged! I also deployed and tested in SIT to make sure everything looked okay.

File: docs/timeseries.md (hunk 1 above)
Comment (nikki-t) [reply docs/timeseries.md:19]: @frankinspace - I missed this last week but I think it is fixed now. Do you think the PR is ready to be merged? Merge conflicts should be resolved as well.

**Review by frankinspace — APPROVED** [review]

**Pull request merged**